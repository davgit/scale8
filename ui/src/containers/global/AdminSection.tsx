import { FC } from 'react';
import { useQuery } from '@apollo/client';
import SignUpApprovalIcon from '../../components/atoms/Icons/SignUpApprovalIcon';
import AdminDashboardIcon from '../../components/atoms/Icons/AdminDashboardIcon';
import NavAdminQuery from '../../gql/queries/NavAdminQuery';
import { NavAdmin } from '../../gql/generated/NavAdmin';
import { buildAdminButtonProps, buildTabButtonProps } from '../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import { toAdmin, toSignupApproval } from '../../utils/NavigationPaths';
import { useRouter } from 'next/router';
import { PageMenuButtonProps } from '../../components/molecules/SideMenuButton';

export const buildAdminTabsMenu = (): PageMenuButtonProps[] => [
    {
        icon: () => <AdminDashboardIcon />,
        label: 'Dashboard',
        link: toAdmin,
    },
    {
        icon: () => <SignUpApprovalIcon />,
        label: 'Signup Approval',
        link: toSignupApproval,
    },
];

const AdminSection: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const router = useRouter();

    const sectionProps: SectionProps<NavAdmin> = {
        // Admin = all permissions
        extractOrgUserDetails: () => ({
            id: '',
            name: '',
            is_paid: false,
            tag_manager_account: {
                id: '',
                trial_expires_in: 0,
                trial_expired: false,
                is_trial: false,
                enabled: true,
            },
            data_manager_account: {
                id: '',
                trial_expires_in: 0,
                trial_expired: false,
                is_trial: false,
                enabled: true,
            },
            me: {
                id: '',
                owner: false,
                permissions: {
                    can_view: true,
                    can_create: true,
                    can_edit: true,
                    can_delete: true,
                    is_admin: true,
                },
            },
        }),
        sectionKey: SectionKey.admin,
        queryResult: useQuery<NavAdmin>(NavAdminQuery),
        buildButtonsProps: (data) => [
            buildAdminButtonProps(router, data.me.orgs),
            buildTabButtonProps(router, buildAdminTabsMenu()),
        ],
        buildMenuItemsProps: () => buildAdminTabsMenu(),
        children: props.children,
    };

    return <Section<NavAdmin> {...sectionProps} />;
};

export default AdminSection;
