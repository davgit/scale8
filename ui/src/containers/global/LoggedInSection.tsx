import * as React from 'react';
import { FC, MouseEventHandler, useEffect, useReducer, useState } from 'react';
import LoggedInTemplate from '../../components/templates/containers/LoggedInTemplate';
import { ApolloError, useQuery } from '@apollo/client';
import { LoggedUser } from '../../gql/generated/LoggedUser';
import LoggedUserQuery from '../../gql/queries/LoggedUserQuery';
import { createHash } from 'crypto';
import { useCancelConfirmDialog } from '../../hooks/dialog/useCancelConfirmDialog';
import { CancelConfirmDialogProps } from '../../types/props/CancelConfirmDialogProps';
import { pageActions } from '../../actions/PageActions';
import { dialogReducer } from '../../context/DialogReducer';
import { orgUserReducer } from '../../context/OrgUserReducer';
import GithubCheck from '../../dialogPages/global/GithubCheck';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import Navigate from '../../components/atoms/Next/Navigate';
import { dialogInit } from '../../context/LoggedInState';
import { useAppContext } from '../../context/AppContext';
import Loader from '../../components/organisms/Loader';
import { configStateFromData } from '../../context/ConfigState';
import { SectionKey } from '../SectionsDetails';
import { clearAuthSession } from '../../utils/authUtils';

export type UserSelectorProps = {
    loading: boolean;
    imgSrc?: string;
    userName: string;
    userEmail: string;
    handleAccountClick: MouseEventHandler;
    handleLogoutClick: MouseEventHandler;
};

export type SideBarProps = {
    handleNotificationClick: MouseEventHandler;
    children?: React.ReactNode;
    userSelectorProps: UserSelectorProps;
    isAdmin: boolean;
    unreadNotifications: number;
};

export type LoggedInProps = {
    sideBarProps: SideBarProps;
    children: React.ReactNode;
    breadcrumb: React.ReactNode;
    sideMenu: React.ReactNode;
    cancelConfirmDialogProps: CancelConfirmDialogProps;
};

const LoggedInSection: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const { setState, loggedInState, configState } = useAppContext();

    const [logOut, setLogOut] = useState(false);

    const [dialogState, dispatchDialogAction] = useReducer(dialogReducer, null, dialogInit);

    const [refreshCurrentPage, setRefreshCurrentPage] = useState(false);
    const [refreshCurrentSection, setRefreshCurrentSection] = useState(false);

    const [snackbarError, setSnackbarError] = useState<ApolloError | null>(null);

    const [orgUserState, dispatchOrgUserAction] = useReducer(orgUserReducer, null);

    const [section, setSection] = useState<symbol>(SectionKey.loggedOut);

    const [sectionHasAnalytics, setSectionHasAnalytics] = useState(false);

    const [gates, dispatchGatesAction] = useReducer(
        (
            state: Record<string, JSX.Element>,
            action: { type: 'set'; payload: { gateName: string; element: JSX.Element } },
        ): Record<string, JSX.Element> => {
            return { ...state, [action.payload.gateName]: action.payload.element };
        },
        {},
    );

    const teleport = (gateName: string, element: JSX.Element) => {
        dispatchGatesAction({ type: 'set', payload: { gateName, element } });
    };

    const { loading, error, data, ...queryResult } = useQuery<LoggedUser>(LoggedUserQuery);

    // Single Question Dialog
    const { dialogProps: cancelConfirmDialogProps, ask } = useCancelConfirmDialog();

    useEffect(() => {
        setState({
            loggedInState: {
                templateInteractions: {
                    ask,
                    dialogState,
                    dispatchDialogAction,
                    snackbarError,
                    setSnackbarError,
                    refreshCurrentPage,
                    setRefreshCurrentPage,
                    refreshCurrentSection,
                    setRefreshCurrentSection,
                    section,
                    setSection,
                    sectionHasAnalytics,
                    setSectionHasAnalytics,
                },
                loggedInUserState: {
                    loggedUserId: data?.me.id ?? '',
                    userIsAdmin: !!data?.me.is_admin,
                    invites: data ? data.me.invites.length : 0,
                    notifications: data ? data.me.user_notifications.length : 0,
                },
                orgUserState,
                dispatchOrgUserAction,
                gates,
                teleport,
            },
            configState: configStateFromData(data),
        });
    }, [
        data,
        dialogState,
        refreshCurrentPage,
        refreshCurrentSection,
        snackbarError,
        orgUserState,
        section,
    ]);

    if (error || logOut) {
        return <Navigate to="/login" />;
    }

    if (loading || loggedInState === undefined || configState === undefined) {
        return <Loader />;
    }

    const gravatarHash = createHash('md5')
        .update(data ? data.me.email.trim().toLowerCase() : '')
        .digest('hex');

    const userSelectorProps: UserSelectorProps = {
        imgSrc: `https://www.gravatar.com/avatar/${gravatarHash}?s=100&d=identicon`,
        loading,
        userName: `${data?.me.first_name} ${data?.me.last_name}`,
        userEmail: `${data?.me.email}`,
        handleAccountClick: () => {
            pageActions.manageAccount({
                dispatchDialogAction,
                refresh: () => {
                    (async () => {
                        await queryResult.refetch();
                    })();
                },
            });
        },
        handleLogoutClick: () => {
            clearAuthSession();
            setLogOut(true);
        },
    };

    const sideBarProps: SideBarProps = {
        handleNotificationClick: () => {
            pageActions.showNotification({
                dispatchDialogAction,
            });
        },
        userSelectorProps,
        isAdmin: data ? data.me.is_admin : false,
        unreadNotifications: data ? data.me.invites.length + data.me.user_notifications.length : 0,
    };

    return (
        <>
            <GithubCheck
                githubUser={data?.me.github_user ?? null}
                githubConnected={data?.me.github_connected ?? false}
            />
            <LoggedInTemplate
                sideBarProps={sideBarProps}
                breadcrumb={gates['breadcrumb'] ?? <></>}
                sideMenu={gates['side'] ?? <></>}
                cancelConfirmDialogProps={cancelConfirmDialogProps}
            >
                {props.children}
            </LoggedInTemplate>
        </>
    );
};

export default LoggedInSection;
