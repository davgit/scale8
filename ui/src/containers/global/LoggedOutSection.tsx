import { FC, useEffect } from 'react';
import LoggedOutTemplate from '../../components/templates/containers/LoggedOutTemplate';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import { clearAuthSession } from '../../utils/authUtils';

const LoggedOutSection: FC<ChildrenOnlyProps & { stayLogged?: boolean }> = ({
    children,
    stayLogged,
}) => {
    useEffect(() => {
        if (!stayLogged) {
            clearAuthSession();
        }
    }, [stayLogged]);

    return <LoggedOutTemplate>{children}</LoggedOutTemplate>;
};

export default LoggedOutSection;
