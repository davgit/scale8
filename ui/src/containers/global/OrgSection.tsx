import { FC } from 'react';
import { useQuery } from '@apollo/client';
import NavOrgQuery from '../../gql/queries/NavOrgQuery';
import { NavOrg } from '../../gql/generated/NavOrg';
import {
    BreadcrumbButtonProps,
    buildOrgButtonProps,
    buildTabButtonProps,
} from '../../utils/BreadcrumbButtonsUtils';
import OrgDashboardIcon from '../../components/atoms/Icons/OrgDashboardIcon';
import OrgUsersIcon from '../../components/atoms/Icons/OrgUsersIcon';
import UserInviteIcon from '../../components/atoms/Icons/UserInviteIcon';
import { Section, SectionItem, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import OrgSettingsIcon from '../../components/atoms/Icons/OrgSettingsIcon';
import { useConfigState } from '../../context/AppContext';
import { NextRouter, useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../types/props/ChildrenAndIdProps';
import { toOrg } from '../../utils/NavigationPaths';
import { PageMenuButtonProps } from '../../components/molecules/SideMenuButton';
import { CurrentOrgPermissions } from '../../context/OrgUserReducer';
import { SvgIconProps } from '@mui/material';

export const buildOrgTabsMenu = (
    orgPermissions: CurrentOrgPermissions,
    id: string,
    useSignup: boolean,
): PageMenuButtonProps[] => {
    return [
        ...[
            {
                icon: () => <OrgDashboardIcon />,
                label: 'Services',
                link: toOrg({ id }, 'dashboard'),
            },
            {
                icon: () => <OrgUsersIcon />,
                label: 'Users',
                link: toOrg({ id }, 'users'),
                disabled: !orgPermissions.isAdmin,
            },
        ],
        ...(useSignup
            ? [
                  {
                      icon: () => <UserInviteIcon />,
                      label: 'User Invites',
                      link: toOrg({ id }, 'invites'),
                      disabled: !orgPermissions.isAdmin,
                  },
              ]
            : []),
        {
            icon: () => <OrgSettingsIcon />,
            label: 'Settings',
            link: toOrg({ id }, 'settings'),
            disabled: !orgPermissions.isAdmin,
        },
    ];
};

export const buildOrgButtons = (
    orgs: SectionItem[],
    currentOrg: SectionItem,
    router: NextRouter,
    orgPermissions: CurrentOrgPermissions,
    useSignup: boolean,
    forceCurrentEntry?: string,
    forceCurrentIcon?: FC<SvgIconProps>,
    isCurrentPage = false,
): BreadcrumbButtonProps[] => [
    buildOrgButtonProps(router, orgs, currentOrg.id, currentOrg.name, isCurrentPage),
    buildTabButtonProps(
        router,
        buildOrgTabsMenu(orgPermissions, currentOrg.id, useSignup),
        forceCurrentEntry,
        forceCurrentIcon,
    ),
];

const OrgSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();

    const { id, children } = props;

    const { useSignup } = useConfigState();

    const sectionProps: SectionProps<NavOrg> = {
        children,
        sectionKey: SectionKey.org,
        extractOrgUserDetails: (data) => data.getOrg,
        queryResult: useQuery<NavOrg>(NavOrgQuery, {
            variables: { id },
        }),
        buildButtonsProps: (data, orgPermissions) => {
            return buildOrgButtons(
                data.me.orgs,
                data.getOrg,
                router,
                orgPermissions,
                useSignup,
                undefined,
                undefined,
                true,
            );
        },
        buildMenuItemsProps: (_, orgPermissions) => {
            return buildOrgTabsMenu(orgPermissions, id, useSignup);
        },
    };

    return <Section<NavOrg> {...sectionProps} />;
};

export default OrgSection;
