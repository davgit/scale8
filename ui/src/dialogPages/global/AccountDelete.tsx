import { FC } from 'react';
import { DialogPageProps } from '../../types/DialogTypes';
import { Box, Button, DialogContent, lighten } from '@mui/material';
import DialogActionsWithCancel from '../../components/molecules/DialogActionsWithCancel';
import { useMutation } from '@apollo/client';
import Loader from '../../components/organisms/Loader';
import { DeleteAccount } from '../../gql/generated/DeleteAccount';
import DeleteAccountQuery from '../../gql/mutations/DeleteAccountQuery';
import FormGqlError from '../../components/atoms/FormGqlError';
import { useLoggedInState } from '../../context/AppContext';
import Navigate from '../../components/atoms/Next/Navigate';
import { toLogin } from '../../utils/NavigationPaths';
import { clearAuthSession } from '../../utils/authUtils';

const AccountDelete: FC<DialogPageProps> = (props: DialogPageProps) => {
    const { templateInteractions } = useLoggedInState();
    const { ask } = templateInteractions;
    const { handleDialogClose } = props;

    const [deleteAccount, { loading, data, error: gqlError }] =
        useMutation<DeleteAccount>(DeleteAccountQuery);

    const successfullySubmitted = data?.deleteMe;

    if (successfullySubmitted) {
        clearAuthSession();
        handleDialogClose(false);
        return <Navigate to={toLogin} />;
    }

    if (loading) {
        return <Loader />;
    }

    return (
        <Box display="flex" flexDirection="column">
            <DialogContent
                sx={{
                    display: 'flex',
                    flexDirection: 'column',
                    minHeight: '248px',
                }}
            >
                <FormGqlError error={gqlError} />
                <small>
                    If you delete your account you will be a automatically removed from all
                    organizations.
                    <br />
                    <br />
                    If you are the owner of an organization you cannot delete your account. you must
                    first transfer the ownership or remove the organization.
                    <br />
                    <br />
                </small>
                <Button
                    variant="contained"
                    onClick={() => {
                        ask(`Are you sure you want to delete your account?`, () => {
                            (async () => {
                                await deleteAccount();
                            })();
                        });
                    }}
                    sx={{
                        color: '#ffffff',
                        backgroundColor: (theme) => lighten(theme.palette.error.main, 0.4),
                        '&:hover': {
                            backgroundColor: (theme) => theme.palette.error.main,
                        },
                    }}
                    disableElevation
                >
                    Delete Account
                </Button>
            </DialogContent>
            <DialogActionsWithCancel handleDialogClose={props.handleDialogClose} />
        </Box>
    );
};

export { AccountDelete };
