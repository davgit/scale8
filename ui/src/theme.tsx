import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
