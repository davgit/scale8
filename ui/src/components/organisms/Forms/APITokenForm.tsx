import makeStyles from '@mui/styles/makeStyles';
import { FC } from 'react';
import { Box, DialogContent } from '@mui/material';
import CopyBlock from '../../atoms/CopyBlock';
import Alert from '@mui/material/Alert';
import DialogActionsWithCancel from '../../molecules/DialogActionsWithCancel';
import { APITokenFormProps } from '../../../dialogPages/global/APIToken';

const useStyles = makeStyles((theme) => ({
    form: {
        display: 'flex',
        flexDirection: 'column',
    },
    submit: {
        margin: theme.spacing(4, 0, 2),
    },
    dialogContent: {
        minHeight: '248px',
    },
}));

const APITokenForm: FC<APITokenFormProps> = (props: APITokenFormProps) => {
    const classes = useStyles();

    return (
        <Box display="flex" flexDirection="column">
            <form className={classes.form} onSubmit={props.handleSubmit}>
                <DialogContent className={classes.dialogContent}>
                    <Box fontSize={12} color="#0000008a">
                        User ID:
                    </Box>
                    <Box mb={1}>
                        <CopyBlock text={props.uid} language="html" flat />
                    </Box>
                    <Box fontSize={12} color="#0000008a">
                        API Token:
                    </Box>
                    <Box mb={1}>
                        <CopyBlock text={props.token} language="html" flat />
                    </Box>
                    <Box mt={3}>
                        <Alert severity="error" style={{ fontSize: '12px' }}>
                            Changing your token will impact any existing API integrations. Please
                            make sure you update your API integrations with your new token.
                        </Alert>
                    </Box>
                </DialogContent>
                <DialogActionsWithCancel
                    handleDialogClose={props.handleDialogClose}
                    confirmText="Refresh Token"
                />
            </form>
        </Box>
    );
};

export default APITokenForm;
