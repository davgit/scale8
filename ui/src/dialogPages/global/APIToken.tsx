import { FC, FormEvent } from 'react';
import { useMutation, useQuery } from '@apollo/client';
import { LoggedUser } from '../../gql/generated/LoggedUser';
import APITokenForm from '../../components/organisms/Forms/APITokenForm';
import LoggedUserQuery from '../../gql/queries/LoggedUserQuery';
import ResetAPITokenQuery from '../../gql/mutations/ResetAPITokenQuery';
import { ResetAPITokenValues } from '../../gql/generated/ResetAPITokenValues';
import { DialogPageProps } from '../../types/DialogTypes';
import { QueryLoaderAndError } from '../../abstractions/QueryLoaderAndError';
import { logError } from '../../utils/logUtils';

type APITokenAfterLoadProps = DialogPageProps & {
    token: string;
    uid: string;
    refetchToken: () => void;
};

export type APITokenFormProps = {
    token: string;
    uid: string;
    handleSubmit: (event?: FormEvent<HTMLFormElement>) => void;
    handleDialogClose: (checkChanges: boolean) => void;
};

const APITokenAfterLoad: FC<APITokenAfterLoadProps> = (props: APITokenAfterLoadProps) => {
    const [resetAPIToken] = useMutation<ResetAPITokenValues>(ResetAPITokenQuery);

    const submitForm = async () => {
        try {
            await resetAPIToken({
                variables: {},
            });
            props.refetchToken();
        } catch (error) {
            logError(error);
        }
    };

    const handleSubmit = (event?: FormEvent<HTMLFormElement>) => {
        if (event) {
            event.preventDefault();
        }

        (async () => {
            await submitForm();
        })();
    };

    return (
        <APITokenForm
            handleSubmit={handleSubmit}
            handleDialogClose={props.handleDialogClose}
            token={props.token}
            uid={props.uid}
        />
    );
};

const APIToken: FC<DialogPageProps> = (props: DialogPageProps) => {
    return QueryLoaderAndError<LoggedUser>(
        false,
        useQuery<LoggedUser>(LoggedUserQuery),
        (data: LoggedUser, valuesRefresh: (mustResetCache: boolean) => void) => {
            return (
                <APITokenAfterLoad
                    token={data.me.api_token}
                    uid={data.me.id}
                    refetchToken={() => valuesRefresh(false)}
                    {...props}
                />
            );
        },
    );
};

export { APIToken };
