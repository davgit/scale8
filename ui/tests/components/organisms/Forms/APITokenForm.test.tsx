import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import APITokenForm from '../../../../src/components/organisms/Forms/APITokenForm';
import { APITokenFormProps } from '../../../../src/dialogPages/global/APIToken';

jest.mock('../../../../src/components/atoms/CopyBlock', () => {
    return () => null;
});

const MockForm: FC = () => {
    const formProps: APITokenFormProps = {
        token: '',
        uid: '',
        handleSubmit: () => {
            //idle
        },
        handleDialogClose: () => {
            //idle
        },
    };

    return <APITokenForm {...formProps} />;
};

describe('APITokenForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
