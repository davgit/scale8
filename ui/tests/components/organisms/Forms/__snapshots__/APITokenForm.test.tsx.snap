// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APITokenForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-j7qwjs"
  >
    <form
      class="MuiBox-root css-mjj15b"
    >
      <div
        class="MuiDialogContent-root css-1pn2u1k-MuiDialogContent-root"
      >
        <div
          class="MuiBox-root css-128pg9j"
        >
          User ID:
        </div>
        <div
          class="MuiBox-root css-19idom"
        />
        <div
          class="MuiBox-root css-128pg9j"
        >
          API Token:
        </div>
        <div
          class="MuiBox-root css-19idom"
        />
        <div
          class="MuiBox-root css-178yklu"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-nygxxl-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-17ii7u1-MuiSvgIcon-root"
                data-testid="ErrorOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-acap47-MuiAlert-message"
            >
              Changing your token will impact any existing API integrations. Please make sure you update your API integrations with your new token.
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-snbeu5-MuiDialogActions-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-7kx4b6-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-18croe1-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Refresh Token
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
